shops/src/main/java/me/mykindos/betterpvp/shops/shops/listeners/ShopListener.java

```diff
@@ -145,6 +155,30 @@ public void onFinalBuyItem(PlayerBuyItemEvent event) {
             event.getGamer().saveProperty(GamerProperty.BALANCE.name(), event.getGamer().getIntProperty(GamerProperty.BALANCE) - cost);
         } else if (event.getCurrency() == ShopCurrency.FRAGMENTS) {
             event.getGamer().saveProperty(GamerProperty.FRAGMENTS.name(), event.getGamer().getIntProperty(GamerProperty.FRAGMENTS) - cost);
+        } else if (event.getCurrency() == ShopCurrency.BARK) {
+            int remaining = cost;
+            Player player = event.getPlayer();
+            ItemStack[] contents = player.getInventory().getContents();
+
+            for (int i = 0; i < contents.length; i++) {
+                ItemStack item = contents[i];
+
+                if (item != null && item.getType() == Material.GLISTERING_MELON_SLICE) {
```

Comment:
You need to check the model data to ensure its the correct item

Better way to do it would be UtilInventory.remove(player, "progression:tree_bark", amount);